relay/submit.go

```diff
@@ -58,6 +58,18 @@ func (rs *Relay) SubmitBlock(ctx context.Context, m *structs.MetricGroup, sbr st
 	}
 	m.AppendSince(tCheckRegistration, "submitBlock", "checkRegistration")
 
+	valErr := make(chan error, 1)
+	go func(ctx context.Context, gasLimit uint64, sbr structs.SubmitBlockRequest, chErr chan error) {
+		defer close(chErr)
+
+		tValidateBlock := time.Now()
+		if err := rs.validateBlock(ctx, gasLimit, sbr); err != nil {
+			chErr <- err
+			return
+		}
+		m.AppendSince(tValidateBlock, "submitBlock", "validateBlock")
```

Comment:
I wonder would this work properly with ObserveWithError. Like as it's async, shouldn't we call observe on that directly

Reply:
I added a lock to `MetricGroup`